src/renderer/components/dock/dock.store.ts

```diff
@@ -185,21 +193,44 @@ export class DockStore implements DockStorageState {
     return reaction(() => [this.height, this.fullSize], callback, options);
   }
 
-  onTabChange(callback: (tabId: TabId) => void, options?: IReactionOptions) {
-    return reaction(() => this.selectedTabId, callback, options);
+  onTabClose(callback: (evt: DockTabCloseEvent) => void, options: IReactionOptions = {}) {
+    return reaction(() => dockStore.tabs.map(tab => tab.id), (tabs: TabId[], prevTabs?: TabId[]) => {
+      if (!Array.isArray(prevTabs)) {
+        return; // tabs not yet modified
+      }
+
+      const closedTabs: TabId[] = prevTabs.filter(id => !tabs.includes(id));
+
+      if (closedTabs.length > 0) {
+        runInAction(() => {
+          closedTabs.forEach(tabId => callback({ tabId }));
+        });
+      }
```

Comment:
nit, we could probably do this unconditionally

```suggestion
      const currentTabs = new Set(tabs => tab.id);
      
      runInAction(() => {
        for (const { id: tabId } of prevTabs {
          if (!currentTabs.has(tabId) {
            callback({ tabId });
          }
        }
      });
```